File: android-app/app/src/main/java/com/vet4pet/app/data/models/Pet.kt
package com.vet4pet.app.data.models

import android.os.Parcelable
import com.vet4pet.app.data.enums.Gender
import kotlinx.parcelize.Parcelize

@Parcelize
data class Pet(
    val id: String,
    val ownerId: String,
    val name: String,
    val species: String,
    val breed: String?,
    val dateOfBirth: Long?,
    val gender: Gender,
    val weight: Double?,
    val profileImageUrl: String?,
    val createdAt: Long
) : Parcelable

File: backend/models/Pet.js
const mongoose = require('mongoose');

const petSchema = new mongoose.Schema(
  {
    name:     { type: String, required: true },
    species:  { type: String },
    breed:    { type: String },
    age:      { type: Number },
    gender:   { type: String },
    ownerId:         { type: String },
    weight:          { type: Number, default: null },
    profileImageUrl: { type: String, default: null },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Pet', petSchema);
